client/post-editor/editor-slug/index.jsx

```diff
@@ -122,7 +122,7 @@ class PostEditorSlug extends Component {
 					<TrackInputChanges onNewValue={ this.recordChangeStats }>
 						<FormTextInput
 							ref="slugField"
-							value={ slug }
+							value={ decodeURI( slug ) }
```

Comment:
Would it solve this issue side wide if we sanitized this slug as it comes from the API and before it gets put in the state tree?